File: src/main/java/implementslegendkt/mod/vaultjp/screen/PlainScreen.java
package implementslegendkt.mod.vaultjp.screen;

import net.minecraft.network.chat.Component;
import net.minecraft.world.inventory.AbstractContainerMenu;

import java.util.List;

public class PlainScreen<M extends AbstractContainerMenu> extends DecentScreen<PlainScreen<M>,M> {


    private final List<Composition<PlainScreen<M>>> compositions;

    protected PlainScreen(M menu, Component p_96550_, List<Composition<PlainScreen<M>>> compositions) {
        super(menu, p_96550_);
        this.compositions = compositions;
    }
}

File: src/main/java/implementslegendkt/mod/screenlegends/PlainScreen.kt
package implementslegendkt.mod.screenlegends

import net.minecraft.network.chat.Component
import net.minecraft.world.inventory.AbstractContainerMenu

class PlainScreen<M : AbstractContainerMenu> (
    menu: M,
    p_96550_: Component?,
    @JvmField val compositions: List<Composition<in PlainScreen<M>>>
) : DecentScreen<PlainScreen<M>, M>(menu, p_96550_)
